packages/components/dataGrid/hooks/useDebouncedDataGridParams.ts

```diff
@@ -0,0 +1,37 @@
+import { useEffect, useState } from 'react';
+import { MRT_PaginationState, MRT_SortingState, MRT_ColumnFiltersState } from 'material-react-table';
+
+import { DataGridRequestParams } from '../types';
+
+import { mapStateToRequestParams } from './utils';
+
+const debounceValue = 100;
+
+export const useDebouncedDataGridRequestParams = (
+  pagination: MRT_PaginationState,
+  sort: MRT_SortingState,
+  filters: MRT_ColumnFiltersState,
+  globalFilter: string,
+) => {
+  const [params, setParams] = useState<DataGridRequestParams>(
+    mapStateToRequestParams(pagination, sort, filters, globalFilter),
+  );
+
+  useEffect(() => {
+    setParams(mapStateToRequestParams(pagination, sort, filters, globalFilter));
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
maybe we should just add missing deps to array instead of adding 
    // eslint-disable-next-line react-hooks/exhaustive-deps ? 
    
    Does this cause any issue?